src/main/java/org/toradocu/translator/ConditionTranslator.java

```diff
@@ -630,6 +630,64 @@ private static String translateSecondPart(String text, DocumentedMethod method)
       translatePropositions(propositions, method);
       conditions.add(propositions.getTranslation());
     }
+
+    //If conditions are empty
+    boolean noSubject = false;
+
+    //Check if it is due to the subject
+    for (PropositionSeries p : extractedPropositions) {
+      noSubject = p.isEmpty();
+      for (Proposition prop : p.getPropositions()) {
+        if (prop.getSubject().getSubject().length() == 0) noSubject = true;
+      }
+    }
```

Comment:
This `for` loop checks that the *last* proposition series contain propositions with no subject. My understanding is that this check enables the addition of the subject when missing. Why do we care only about the last proposition series?